src/js/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,207 @@
+import React, { forwardRef, useContext, useState } from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { Box } from '../Box';
+import { Nav } from '../Nav';
+import { PageIndex } from './PageIndex';
+import { usePagination } from '../../utils/pagination';
+
+const StyledPaginationContainer = styled(Box)`
+  ${props =>
+    props.theme.pagination.container && props.theme.pagination.container.extend}
+`;
+
+const Pagination = forwardRef(
+  (
+    {
+      a11yTitle,
+      numItems,
+      numEdgePages = 1, // number of pages at each edge of page indices
+      // number of page controls in the middle
+      numMiddlePages: numMiddlePagesProp = 3,
+      onChange,
+      page: pageProp,
+      step: stepProp,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+
+    const [, , defaultPage, step] = usePagination({
```

Comment:
This looks like we're trying to put too much into `usePagination`. We already have the `step`. Do we even need `defaultPage` via this method?

Reply:
Thanks, I've cleaned up.